src/main.c

```diff
@@ -1007,23 +1008,41 @@ static void free_status_print(RaucStatusPrint *status)
 
 G_DEFINE_AUTOPTR_CLEANUP_FUNC(RaucStatusPrint, free_status_print);
 
+/* Definition list for terminal colors */
+#define KNRM  "\x1B[0m"
+#define KRED  "\x1B[31m"
+#define KGRN  "\x1B[32m"
+#define KYEL  "\x1B[33m"
+#define KBLU  "\x1B[34m"
+#define KMAG  "\x1B[35m"
+#define KCYN  "\x1B[36m"
+#define KWHT  "\x1B[37m"
+#define KBLD  "\x1B[1m"
+
 static void r_string_append_slot(GString *text, RaucSlot *slot, RaucStatusPrint *status)
 {
 	RaucSlotStatus *slot_state = slot->status;
 
-	g_string_append_printf(text, "\n  %s: class=%s, device=%s, type=%s, bootname=%s\n",
-			slot->name, slot->sclass, slot->device, slot->type, slot->bootname);
-	g_string_append_printf(text, "      state=%s, description=%s", r_slot_slotstate_to_str(slot->state), slot->description);
-	if (slot->parent)
-		g_string_append_printf(text, ", parent=%s", slot->parent->name);
-	else
-		g_string_append(text, ", parent=(none)");
-	if (slot->mount_point || slot->ext_mount_point)
-		g_string_append_printf(text, ", mountpoint=%s", slot->mount_point ? slot->mount_point : slot->ext_mount_point);
-	else
-		g_string_append(text, ", mountpoint=(none)");
+	setlocale(LC_ALL, "");
+
+	g_string_append_printf(text, "%lc %s%s%s"KBLD "[%s]"KNRM " (%s, %s, %s%s"KWHT ")"KNRM,
+			slot->bootname ? slot == status->primary ? L'\u23FA' :  L'\u25CB' : ' ',
```

Comment:
I'm not sure that `%lc` and L'\23FA' is the correct approach. There is g_string_append_unichar() which could be used instead. As this is at the beginning of this printf, it should be simple to adapt.

Reply:
The i386 build also fails with
```
src/main.c:1028:34: error: format '%lc' expects argument of type 'wint_t', but argument 3 has type 'long int' [-Werror=format=]
```

Reply:
Yes, this is probably also more obvious. I change it to use `g_string_append_unichar()`.